chapter09/ard/ex2.pl

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use v5.12;
+
+use File::Slurp;
+
+my @lines = read_file( 'patterns' );
+my @patterns;
+
+foreach my $line ( @lines ) {
+  chomp $line;
+  my $pattern;
+  eval {
+    $pattern = qr/$line/
+  };
+  if ( $@ ) {
+    warn "Invalid pattern: $@";
+    next;
+  }
+  else {
+    push @patterns, $pattern;
+  }
```

Comment:
You use `next` in the error condition, so this doesn't need to be in an else body. Alternatively, you could remove the `next`.